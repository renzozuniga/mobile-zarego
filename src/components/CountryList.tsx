import React, { useState, useEffect } from 'react';
import Axios from "axios";
import {
    FlatList,
    TouchableOpacity,
    View,
    Text,
    Pressable
} from 'react-native';
import { REACT_MOBILE_BACKEND_ZAREGO } from "@env";

const initialPage = `${REACT_MOBILE_BACKEND_ZAREGO}/api/countries/options`;

const CountryList = ({ navigation }: any) => {
    const [options, setOptions] = useState([]);
    const [selected, setSelected] = useState<number[]>([]);

    useEffect(() => {
        Axios.get(initialPage)
            .then((response: any) => {
                setOptions(response.data.data);
            });
    }, []);

    const onSelect = (id: any) => {
        setSelected((oldArray: any) => {
            if (oldArray.includes(id)) {
                return oldArray.filter((el: any) => el !== id);
            } else {
                return [...oldArray, id];
            }
        });
    }

    return (
        <View style={{ flex: 1 }}>
            <Text style={{ textAlign: 'center', margin: 20, fontWeight: 'bold', fontSize: 24 }}>CHOOSE COUNTRIES</Text>
            <View style={{ marginLeft: 20, marginRight: 20 }}>
                <FlatList
                    data={options}
                    renderItem={({ item, index }: any) => {
                        return (
                            <TouchableOpacity style={{
                                width: '100%',
                                height: 70,
                                borderColor: '#8e8e8e',
                                borderWidth: 0.2,
                                justifyContent: 'center',
                                alignItems: 'center',
                                backgroundColor: selected.includes(item.value) ? 'rgb(155 195 240)' : 'white'
                            }}
                                onPress={() => {
                                    onSelect(item.value);
                                }}>
                                <Text style={{ fontSize: 20 }}>{item.label}</Text>
                            </TouchableOpacity>
                        )
                    }}
                    style={{ height: '70%' }}
                />
            </View>
            <View style={{
                position: 'absolute',
                bottom: '5%',
                width: '60%',
                left: '20%',
                right: '20%',
                backgroundColor: 'white',
                justifyContent: 'center',
                alignItems: 'center',
                borderWidth: 1,
                borderColor: "black",
                borderRadius: 10,
            }}>
                <TouchableOpacity style={{
                    margin: 5,
                    backgroundColor: 'white',
                    borderRadius: 8,
                    height: 50,
                    width: '60%',
                    alignItems: 'center',
                    flexDirection: 'row'
                }} >
                    <View style={{ backgroundColor: 'white', height: 'auto', width: '100%' }}>
                        <Pressable style={{
                            alignItems: 'center',
                            justifyContent: 'center',
                            paddingVertical: 12,
                            paddingHorizontal: 12,
                            borderRadius: 4,
                            elevation: 3,
                            backgroundColor: 'white',
                        }} onPress={() => navigation.navigate("CountryTable", { list: selected })}>
                            <Text style={{
                                fontSize: 20,
                                lineHeight: 21,
                                fontWeight: 'bold',
                                letterSpacing: 0.25,
                                color: 'black',
                            }}>See Data</Text>
                        </Pressable>
                    </View>
                </TouchableOpacity>
            </View>
        </View >
    );
};

export default CountryList;