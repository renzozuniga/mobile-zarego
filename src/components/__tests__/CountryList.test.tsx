import React from 'react'
import TestRenderer from 'react-test-renderer';
import CountryList from '../CountryList';

jest.useFakeTimers()

describe('src/components/CountryList', () => {
    it('renders correctly', () => {
        const instance = TestRenderer.create(<CountryList />).toJSON();
        expect(instance).toMatchSnapshot();
    })

    it('renders sections list', () => {
        const instance = TestRenderer.create(<CountryList />).toJSON();
        expect(instance.children.length).toBe(3);
    })

    it('renders title', async () => {
        const instance = TestRenderer.create(<CountryList />).toJSON();
        const text = instance.children[0];
        expect(text.children[0]).toBe('CHOOSE COUNTRIES');
    })

    it('renders flatlist', async () => {
        const instance = TestRenderer.create(<CountryList />).toJSON();
        const flat = instance.children[1].children[0];
        expect(flat.props.data.length).toBe(0);
    })

    it('renders button', async () => {
        const navigate = jest.fn();
        const instance = TestRenderer.create(<CountryList navigation={{ navigate }} />).toJSON();
        const btn = instance.children[2].children[0].children[0].children[0];
        btn.props.onClick();
        expect(btn.children.length).toBe(1);
    })
})