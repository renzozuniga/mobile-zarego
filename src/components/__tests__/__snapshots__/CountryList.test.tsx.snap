// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/CountryList renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    CHOOSE COUNTRIES
  </Text>
  <View
    style={
      {
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      style={
        {
          "height": "70%",
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "black",
        "borderRadius": 10,
        "borderWidth": 1,
        "bottom": "5%",
        "justifyContent": "center",
        "left": "20%",
        "position": "absolute",
        "right": "20%",
        "width": "60%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 8,
          "flexDirection": "row",
          "height": 50,
          "margin": 5,
          "opacity": 1,
          "width": "60%",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "height": "auto",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 4,
              "elevation": 3,
              "justifyContent": "center",
              "paddingHorizontal": 12,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontSize": 20,
                "fontWeight": "bold",
                "letterSpacing": 0.25,
                "lineHeight": 21,
              }
            }
          >
            See Data
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
